verification/src/solidity/verifier.rs

```diff
@@ -97,3 +327,329 @@ impl Verifier {
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            DEFAULT_CREATION_TX_INPUT,
+            invalid_input,
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
+        assert_eq!(
+            verifier.unwrap_err(),
+            InitializationError::InvalidDeployedBytecode
+        )
+    }
+
+    #[test]
+    fn test_initialization_with_metadata_hash_mismatch_should_fail() {
+        // {"ipfs": h'1220EB23CE2C13EA8739368F952F6C6A4B1F0623D147D2A19B6D4D26A61AB03FCD3E', "solc": 0.8.0}
+        let another_metadata_hash = "a2646970667358221220eb23ce2c13ea8739368f952f6c6a4b1f0623d147d2a19b6d4d26a61ab03fcd3e64736f6c63430008000033";
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            &format!(
+                "{}{}",
+                DEFAULT_BYTECODE_WITHOUT_METADATA_HASH, another_metadata_hash
+            ),
+            DEFAULT_DEPLOYED_BYTECODE,
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
+        assert_eq!(
+            verifier.unwrap_err(),
+            InitializationError::MetadataHashMismatch(Mismatch::expected(
+                DisplayBytes::from_str(DEFAULT_ENCODED_METADATA_HASH).unwrap()
+            ))
+        );
+    }
+}
+
+#[cfg(test)]
+mod metadata_hash_deserialization_tests {
+    use super::*;
+
+    fn is_valid_custom_error(
+        error: minicbor::decode::Error,
+        expected: ParseMetadataHashError,
+    ) -> bool {
+        if !error.is_custom() {
+            return false;
+        }
+
+        // Unfortunately, current `minicbor::decode::Error` implementation
+        // does not allow to retrieve insides out of custom error,
+        // so the only way to ensure the valid error occurred is by string comparison.
+        let parse_metadata_hash_error_to_string = |err: ParseMetadataHashError| match err {
+            ParseMetadataHashError::NonExhausted => "NonExhausted",
+            ParseMetadataHashError::InvalidSolcType(_) => "InvalidSolcType",
+            ParseMetadataHashError::DuplicateKeys => "DuplicateKeys",
+        };
+        format!("{:?}", error).contains(parse_metadata_hash_error_to_string(expected))
+    }
+
+    #[test]
+    fn test_deserialization_metadata_hash_without_solc_tag() {
+        // given
+        // { "bzzr0": b"d4fba422541feba2d648f6657d9354ec14ea9f5919b520abe0feb60981d7b17c" }
+        let hex =
+            "a165627a7a72305820d4fba422541feba2d648f6657d9354ec14ea9f5919b520abe0feb60981d7b17c";
+        let encoded = DisplayBytes::from_str(hex).unwrap().0;
+        let expected = MetadataHash { solc: None };
+
+        // when
+        let decoded =
+            MetadataHash::from_cbor(encoded).expect("Error when decoding valid metadata hash");
+
+        // then
+        assert_eq!(expected, decoded, "Incorrectly decoded");
+    }
+
+    #[test]
+    fn test_deserialization_metadata_hash_with_solc_as_version() {
```

Comment:
is github lagging or you forgot to push some commits or didn't you removed the `test` from test names? https://github.com/blockscout/blockscout-rs/pull/8#discussion_r895070780